src/lib/operators/aggregate_sort.cpp

```diff
@@ -379,23 +379,44 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
    *               So in total, group_boundaries will contain one element less than there are groups.
    */
   std::set<RowID> group_boundaries;
-  auto chunks = sorted_table->chunks();
+  const auto& chunks = sorted_table->chunks();
   for (const auto& column_id : _groupby_column_ids) {
     auto data_type = input_table->column_data_type(column_id);
     resolve_data_type(data_type, [&](auto type) {
       using ColumnDataType = typename decltype(type)::type;
       std::optional<ColumnDataType> previous_value;
       ChunkID chunk_id{0};
+
+      /*
+       * Initialize previous_value to the first value in the table, so we avoid considering it a value change.
+       * We do not want to consider it as a value change, because we the first row should not be part of the boundaries.
+       * For the reasoning behind it see above.
+       */
+      const auto& first_segment = chunks[0]->get_segment(column_id);
+      auto pos_list = std::make_shared<PosList>();
+      pos_list->emplace_back(RowID{ChunkID{0}, 0});
+      pos_list->guarantee_single_chunk();
+      segment_iterate_filtered<ColumnDataType>(*first_segment, pos_list, [&](const auto& position) {
+        if (position.is_null()) {
+          previous_value.reset();
+        } else {
+          previous_value.emplace(position.value());
+        }
+      });
```

Comment:
In this particular case, using `operator[]` is probably more efficient (and definitely more readable) than going through all this trouble.

Reply:
Okay, we'll change it and add a comment that we are aware that we usually should avoid `operator[]` , but `segment_iterate_filtered` would not be more efficient here.